runtime/compiler/trj9/optimizer/JProfilingValue.cpp

```diff
@@ -204,6 +206,46 @@ TR_JProfilingValue::perform()
    }
 
 void
+TR_JProfilingValue::removeRedundantProfilingValueCalls()
+   {
+   TR::TreeTop *cursor = comp()->getStartTree();
+   TR_BitVector *alreadyProfiledValues = new (comp()->trStackMemory()) TR_BitVector();
+   while (cursor)
+      {
+      TR::Node * node = cursor->getNode();
+      if (node->isProfilingCode() &&
+         node->getOpCodeValue() == TR::treetop &&
+         node->getFirstChild()->getOpCode().isCall() &&
+         (node->getFirstChild()->getSymbolReference()->getReferenceNumber() == TR_jProfile32BitValue ||
+         node->getFirstChild()->getSymbolReference()->getReferenceNumber() == TR_jProfile64BitValue) &&
+         node->getFirstChild()->getNumChildren() == 3)
+         {
+         TR::Node *value = node->getFirstChild()->getFirstChild();
+         if (alreadyProfiledValues->isSet(value->getGlobalIndex()))
+            {
+            /* Found that second and third child were having more than one ref counts. 
+            TR_ASSERT_FATAL(node->getFirstChild()->getSecondChild()->getReferenceCount() == 1 &&
+               node->getFirstChild()->getThirdChild()->getReferenceCount() == 1,
+               "Second and Third Child of the value calls should be referenced only once");
+            */
+            performTransformation(comp(), "%s Removing profiling treetop, node n%dn is already profiled\n",
```

Comment:
this needs to be in an if so that limiting transformations works since removing the tree is optional.